### PR TITLE
Correct transaction dependency ordering on startup mempool resync

### PR DESCRIPTION
**Pull request by cjepson**
Submission of unconfirmed transactions to daemon memory pool upon connection
would sometimes fail because their dependency ordering was incorrect. This
sorts them as a directed acyclic graph so that their ordering doesn't cause
'unknown inputs' errors upon submission to the daemon memory pool.

Corrects issue #156

### CODE REVIEW

**Comment by jrick** [comment]
wire.ShaHash has value equality using == so just use that instead of IsEqual (and this matches the rest of the codebase).

Slices can be ranged over without first checking for nil (a nil slice is treated the same as a non-nil slice with 0 length).

We should error for cycles since that should only happen if you pass in invalidly signed transactions.

File: wtxmgr/unconfirmed.go [hunk 1]
```diff
@@ -164,9 +164,51 @@ func (s *Store) unminedTxs(ns walletdb.Bucket) ([]*wire.MsgTx, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	order, _, err := topSortKahn(g, i)
+	if err != nil {
+		return nil, err
+	}
+
+	// Transactions with no local depencies are excluded from this list, so
```
Comment (jrick) [inline wtxmgr/unconfirmed.go]: typo: depencies

**Comment by cjepson** [comment]
Recommended changes have been made, with the exception of adding an error for cycle detection. Cycles are already detected and an error is returned if they are found.

**Comment by jrick** [comment]
I'd like to get this in.  Unfortunately it looks like the webpage containing the original implementation is covered under the GNU Free Documentation License, and I'm unsure if we can use the code there without violating the GPL (or any GNU copyleft license).

**Comment by jrick** [comment]
I've reimplemented this change without copying the code from another source due to the license concerns.  Closing this PR and will open a new one.

**Pull request closed without merging**